python/lsst/pipe/supertask/super_task.py

```diff
@@ -101,119 +44,72 @@ class SuperTask(Task):
-    prepared to be parsed to run() which actually
-    performs the operation
-
-    The name is taken from _default_name (a more pythonic way to deal with variable names) but is
-    still backwards compatible with Task.
+    prepared to be parsed to run() which actually performs the operation
 
     Ideally all Task should be (and will be) SuperTasks, which provides the abstraction and
     expose the run method of the Task itself.
     """
 
-    _default_name = None
-    _parent_name = None
-
-    def __init__(self, config=None, name=None, parent_task=None, log=None, activator=None, butler=None):
+    def __init__(self, config=None, name=None, parent_task=None, log=None, butler=None):
         """
-        Creates the SuperTask, the parameters are the same as Task, except by activator which is a
-        hook for the class activator that calls this Task, for cmdLineActivator is the only one
-        available
+        Creates the SuperTask, the parameters are the same as Task.
 
-          The inputs are (some of them taken from task.py):
+        The inputs are (some of them taken from task.py):
 
         :param config:      configuration for this task (an instance of self.ConfigClass,
                             which is a task-specific subclass of lsst.pex.config.Config), or None.
                             If None:
                               - If parentTask specified then defaults to parentTask.config.name
                               - If parentTask is None then defaults to self.ConfigClass()
         :param name:        brief name of super_task, or None; if None then defaults to
-                            self._default_name or self._DefaultName
+                            self._DefaultName
         :param parent_task: the parent task of this subtask, if any.
                              - If None (a top-level task) then you must specify config and name is
                                ignored.
                              - If not None (a subtask) then you must specify name
-        :param log:         pexLog log; if None then the default is used;
-                            in either case a copy is made using the full task name.
-        :param activator:   name of activator calling this task, default is None,
-                            there is only one kind of activator (cmdLineActivator)
+        :param log:         log (an lsst.log.Log) whose name is used as a log name prefix,
+                            or None for no prefix. Ignored if parentTask specified, in which case
+                            parentTask.log's name is used as a prefix.
+        :param butler:      data butler instance (this is not used by this class, it should
+                            probably be removed)
         :return: The SuperTask Class
         """
 
+        # No spaces allowed in names for SuperTasks
+        # TO DO: need better validation here, some other characters may be disallowed
```

Comment:
some of the taks names in pipe_tasks had `_` but you are right, a more generic validation is needed.
something like this?
```python
invalid_chars = ['-','$','~','@','*','_']`
if any((char in invalid_chars) for char in name):  raise NameException
```